### PR TITLE
[adding] ADR-12: JetStream encryption at rest

### PR DESCRIPTION
**Pull request by ripienaar**
Signed-off-by: R.I.Pienaar <rip@devco.net>

### CODE REVIEW

**Comment by ripienaar** [comment]
This just captures the internal google doc into an ADR.

@derekcollison if you dont mind checking this - I updated the bits about AES etc, since it seems in the PR we just support cha cha polly.

**Comment by ripienaar** [comment]
Relates to https://github.com/nats-io/nats-architecture-and-design/issues/15

File: adr/ADR-12.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+## Keys
+
+ * *EK* - External Key, provided to the nats-server configuration or via encryption service in subsequent releases.
+ * *KEK* - An encryption key derived from the *EK*. Always generated and never stored. This is generated via a PRF where 
+   `KEK := PRF(EK, IV, Context)`, e.g. `HMAC-SHA-256`, `Argon2Id`
+ * *AEK* - Asset encryption key, encrypted via a *KEK* and placed alongside the asset on disk.
+
+## Encryption
+
+Encryption will utilize an authenticated encryption AEAD. We use ChaCha20-Poly1305.
```
Comment (ripienaar) [inline adr/ADR-12.md:31]: this is the only line I updated from the google doc @derekcollison

**Comment by derekcollison** [comment]
On my todo list.

File: adr/ADR-11.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## Encryption
+
+Encryption will utilize an authenticated encryption AEAD. We use ChaCha20-Poly1305.
+
+## Future Support
+
+### KMS
+
+In a follow-on release we may choose to configure the nats-servers to access a KMS and not have direct access to the *EK*.
```
Comment (derekcollison) [inline adr/ADR-11.md:37]: For now we expect folks to use NATS configuration variables to give access to the master key to a server.

**Review by derekcollison — APPROVED** [review]
LGTM - Small comment.

**Pull request merged**